### PR TITLE
issue-25133: adding background grid guide

### LINKED ISSUE #25133: Template Builder: Adding a Background Grid Guide for 12 Columns with Gaps
### User Story
As a dotCMS user, I want to be able to have grids as guidelines to resize and move boxes and have a better understanding of the layout.

### Acceptance Criteria
- The background grid should be responsive.
- The background grid should use CSS gradients.
- The grid needs to match the width of the boxes

### Proposed Objective
Core Feature

### Proposed Priority
Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.:
[Figma](https://www.figma.com/file/ECuBLIRWZjEzLzmlPrA0Rk/Template-Builder?type=design&node-id=358%3A25044&t=KBNJK64CaImeWYfV-1)

### Assumptions & Initiation Needs
- The layout should be tested in all browsers and devices.

### Quality Assurance Notes & Workarounds
- The background grid should correctly represent the 12 columns with gaps.

### Sub-Tasks & Estimates:
- Create the CSS gradients for the background grid
- Test the layout in all browsers and devices down to iPad landscape

### PR DESCRIPTION
**Pull request by danielmdob**
### <samp>🤖 Generated by Copilot at f6b5be1</samp>

This pull request introduces a new component for the template builder feature, which allows users to customize the background columns of a template. It refactors the template builder component to use this new component and updates the styles and stories accordingly. It also adds a basic test for the new component. The affected files are `template-builder.component.html`, `template-builder.component.scss`, `template-builder.component.stories.ts`, `template-builder.module.ts`, and the files in the `template-builder-background-columns` folder.

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
image  |  image

### CODE REVIEW

File: core-web/libs/template-builder/src/lib/components/template-builder/components/template-builder-background-columns/template-builder-background-columns.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { TemplateBuilderBackgroundColumnsComponent } from './template-builder-background-columns.component';
+
+describe('TemplateBuilderBackgroundColumnsComponent', () => {
```
Comment (danielmdob) [inline core-web/libs/template-builder/src/lib/components/template-builder/components/template-builder-background-columns/template-builder-background-columns.component.spec.ts:5]: todo: add tests using spectator if this approach is good to go

**Review by jdotcms — APPROVED** [review]

**Comment by danielmdob** [comment]
Pretty sure coverage check should have passed, don't know what happened.

**Comment by fmontes** [comment]
image

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/libs/template-builder/src/lib/components/template-builder/components/template-builder-background-columns/template-builder-background-columns.component.scss [hunk 2]
```diff
@@ -0,0 +1,16 @@
+@use "variables" as *;
+
+.background-columns {
+    height: 100%;
+    width: 100%;
+    grid-template-columns: repeat(12, minmax(0, 1fr));
+    display: grid;
+    left: 3.7rem;
```
[the remark this answers is not in the document]
Comment (danielmdob) [reply core-web/libs/template-builder/src/lib/components/template-builder/components/template-builder-background-columns/template-builder-background-columns.component.scss]: Seems difficult as there is no element that extends to something proportional to the maximum possible width of the boxes at that level.  What do you think of going for a padding only approach and putting it in a constant as I did in the last commit?

File: core-web/libs/template-builder/src/lib/components/template-builder/utils/gridstack-options.ts [hunk 3]
```diff
@@ -32,7 +38,7 @@ function isARowWidget(el: Element): boolean {
 export const subGridOptions: GridStackOptions = {
     cellHeight: 224,
     column: 'auto',
-    margin: 16,
+    margin: `${GRID_STACK_MARGIN}${GRID_STACK_UNIT}`,
```
Comment (zJaaal) [inline core-web/libs/template-builder/src/lib/components/template-builder/utils/gridstack-options.ts:41]: This should be 1 rem

**Review by zJaaal — CHANGES_REQUESTED** [review]

**Pull request merged**